sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ExprUtils.scala

```diff
@@ -169,7 +169,11 @@ object ExprUtils extends QueryErrorsBase {
         a.failAnalysis(
           errorClass = "MISSING_GROUP_BY",
           messageParameters = Map.empty)
-      case e: Attribute if !a.groupingExpressions.exists(_.semanticEquals(e)) =>
+      case e: Attribute if !a.groupingExpressions.exists {
```

Comment:
do you need this? afaik you are doing collationkey insertion only there is no string in aggregate?

Reply:
correct, this is not needed - we won't need to "match" aggregate and grouping expressions with collationKey, since we only support collationKey injection for "pure" grouping aggregation on collated string columns